podaac/subsetter/subset.py

```diff
@@ -978,12 +978,15 @@ def open_as_nc_dataset(filepath: str) -> Tuple[nc.Dataset, bool]:
         nc_dataset, has_groups = h5file_transform(filepath)
     else:
         # Open dataset with netCDF4 first, so we can get group info
-        nc_dataset = nc.Dataset(filepath, mode='r')
-        has_groups = bool(nc_dataset.groups)
+        try:
+            nc_dataset = nc.Dataset(filepath, mode='r')
+            has_groups = bool(nc_dataset.groups)
 
-        # If dataset has groups, transform to work with xarray
-        if has_groups:
-            nc_dataset = transform_grouped_dataset(nc_dataset, filepath)
+            # If dataset has groups, transform to work with xarray
+            if has_groups:
+                nc_dataset = transform_grouped_dataset(nc_dataset, filepath)
+        except OSError:
```

Comment:
This seems a tad broad, is there a better way to identify this situation?

Reply:
hmmm how can we identify hdf5.... we have been using extension but this one had .nc for aquarius

Reply:
Reading the first 8 bytes of the file would probably be the most accurate (it should be a static value as described https://web.ics.purdue.edu/~aai/HDF5/html/H5.format.html) 

But maybe that's a bit overkill. 

Is the `nc_dataset = nc.Dataset(filepath, mode='r')` line the one that throws the OSError if it can't be opened as netcdf? Maybe we can just put the try around that one line instead?

Reply:
its in the transform_grouped_dataset function where we try to open the file in append mode. nc_dataset = nc.Dataset(file_to_subset, 'r+'). I can try other things too i read i can read the first few bytes of the file to identify HDF5

Reply:
I have tried reading the header, it seems like netcdf 4 classic has the same hdf5 headers so reading the headers didn't work.

Reply:
I'm just going to approve this and get it merged.  We can open a new ticket to improve this later when we come up with a better solution besides try except.